cmake_modules/ParsecCompilerFlags.cmake

```diff
@@ -60,9 +62,9 @@ function(check_and_set_compiler_option)
   else()
     set(${parsec_cas_co_NAME} "${parsec_cas_co_defined_options}")
   endif()
-  if(DEFINED ${parsec_cas_co_NAME})
-    add_compile_options("${${parsec_cas_co_NAME}}")
-  endif()
+  #  if(DEFINED ${parsec_cas_co_NAME})
```

Comment:
Doesn't that disable the whole thing? That won't work when using important, non-optional flags like -mcx16

Reply:
In theory (reading the function) the option that is added with add_compile_options should have generator expressions to protect against passing the wrong file to the wrong compiler, that is, the string in `parsec_cas_co_NAME` should have only things like `$<$<LANGUAGE:C>:-fcolor-diagnostics>`, which then should expand to nothing when the compile rule applies with Fortran.

Reply:
The PR #326 was wrong, my apologies for not seeing that... I created PR #336 that fixes those issues and take some code from that PR to sanitize the messages.

I have tested with an heterogeneous setup (icc + gfortran + g++), because icc accepts the -WdXXXX, but not g++ or gfortran, and this time it seems to work, but if you can test on your mac setup that would be a good validation.